xarray/conventions.py

```diff
@@ -389,6 +389,15 @@ def decode_cf_variable(
     if decode_times:
         var = times.CFDatetimeCoder(use_cftime=use_cftime).decode(var, name=name)
 
+    # transform numpy object-dtype strings to numpy unicode strings
```

Comment:
I'm not absolutely sure, if this is really what we want. This cast's the variable to the _expected_ dtype, but I can't immediately see the implications for users downstream.

Reply:
removed from PR